server/order/controller/product_controller.js

```diff
@@ -5,8 +5,17 @@
 const Product = require('../model/product');
 
 exports.addProduct = function (productData) {
-  const product = new Product(JSON.parse(productData));
-  product.save();
+  return new Promise((resolve, reject) => {
+    Product.findOneAndUpdate({ name: productData.name }
```

Comment:
I think there are no product that has same name. aren't you?
And It should use findOneAndUpdate with upsert option because of can't has same name product.